frontend/src/utils/dialogs/FarmWithdrawDialog.vue

```diff
@@ -130,14 +130,52 @@ export default class FarmWithdrawDialog extends Vue {
 
 	async zapOut() {
 		const zap: WBANFarmZap = WBANFarmZap__factory.connect(getZapAddress(), this.signer)
+		const lpAddress = this.farm.lpAddresses[FarmWithdrawDialog.ENV_NAME as keyof Address]
+		const lpPair = await IUniswapV2Pair__factory.connect(lpAddress, this.signer)
+
 		const amountToZap = parseEther(this.lpAmount)
 
-		const lpAddress = this.farm.lpAddresses[FarmWithdrawDialog.ENV_NAME as keyof Address]
-		const erc20 = await ERC20PermitUpgradeable__factory.connect(lpAddress, this.signer)
-		const nonce = await erc20.nonces(this.account)
+		const weth = await TokensUtil.getTokenBySymbol('WETH')
+		if (!weth) {
+			console.error("Can't find WETH")
+			return
+		}
+		const tokenAddress = this.token.address ? this.token.address : weth[0].address
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const token: string = this.farm.token!.address![FarmWithdrawDialog.ENV_NAME as keyof Address]
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const quoteToken: string = this.farm.quoteToken!.address![FarmWithdrawDialog.ENV_NAME as keyof Address]
+		const otherTokenAddress = tokenAddress.toLowerCase() === token.toLowerCase() ? quoteToken : token
+
+		const router = IUniswapV2Router02__factory.connect(await zap.router(), this.signer)
+		const reserves = await lpPair.getReserves()
+		console.debug('Reserves:', formatEther(reserves.reserve0), formatEther(reserves.reserve1))
+		const lpTotalSupply = await lpPair.totalSupply()
+		console.debug('LP Total supply', formatEther(lpTotalSupply))
+		const isToken0 = (await lpPair.token0()).toLowerCase() === tokenAddress.toLowerCase()
+		const amountIn = (isToken0 ? reserves.reserve1 : reserves.reserve0)
+			.mul(parseEther(this.lpAmount))
+			.div(lpTotalSupply)
+		console.debug('amountIn', formatEther(amountIn), otherTokenAddress)
+		let [, swapAmountOut] = await router.getAmountsOut(amountIn, [otherTokenAddress, this.token.address])
```

Comment:
This works because the route only use the two tokens, good.

Reply:
Indeed. We are swapping right from the liquidity pool pair.